olive/platform_sdk/qualcomm/env.py

```diff
@@ -0,0 +1,98 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import os
+import platform
+from pathlib import Path
+
+from olive.platform_sdk.qualcomm.constants import SDKTargetDevice
+
+
+class SDKEnv:
+    def __init__(self, sdk: str, root_env_name: str, target_arch: str = None, dev: bool = False):
+        self.sdk = sdk
+        self.root_env_name = root_env_name
+        # dev: Whether to use the development version of the SDK,
+        # only applicable to SNPE/QNN x86_64 linux/windows
+        self.dev = dev
+        self.target_arch = self._verify_target_arch(target_arch) if target_arch else self._infer_target_arch()
+        self.delimiter = os.path.pathsep
+
+    @property
+    def sdk_root_path(self) -> str:
+        sdk_root = os.environ.get(self.root_env_name)
+        if not sdk_root:
+            raise ValueError(f"{self.root_env_name} is not set")
+        return sdk_root
+
+    def _verify_target_arch(self, target_device: SDKTargetDevice) -> str:
+        if not self.root_env_name:
+            raise FileNotFoundError(f"Path {self.sdk_root_path} does not exist")
```

Comment:
this should be part of validation of root_env_name assignment. 

something like in __init__.py
if not root_env_name:
     raise ValueError(xxx)

Reply:
Updated.